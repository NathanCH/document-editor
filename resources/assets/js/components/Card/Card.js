import React from 'react';
import PropTypes from 'prop-types';

import { 
  Card,
  CardBody,
  CardTitle,
  CardText
} from 'reactstrap';

import './Card.scss';

const CardWrapper = props => (
  <Card className={props.className}>
    {props.customSection}
    <CardBody>
      <CardTitle>{props.title}</CardTitle>
      <CardText>{props.text}</CardText>
    </CardBody>
  </Card>
);

CardWrapper.propTypes = {
  className: PropTypes.string,
  customSection: PropTypes.element,
  title: PropTypes.string,
  text: PropTypes.string,
};

CardWrapper.defaultProps = {
  className: '',
  customSection: null,
  title: null,
  text: null,
};

export default CardWrapper;