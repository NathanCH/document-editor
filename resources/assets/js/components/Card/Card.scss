@import './resources/assets/sass/app';

.card-title {
  color: $body-color;
  font-weight: $font-weight-bold;
}

.card-text {
  color: $text-muted;
}