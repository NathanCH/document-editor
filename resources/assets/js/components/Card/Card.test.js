import React from 'react';
import renderer from 'react-test-renderer';

import Card from './Card';

describe('<Card />', () => {
  test('it renders without props', () => {
    const component = renderer
      .create(<Card />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  test('it renders `props.customSection`', () => {
    const SubComponent = () => <span>Test</span>;

    const component = renderer
      .create(<Card customSection={<SubComponent />} />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  test('it renders `props.title`', () => {
    const component = renderer
      .create(<Card title="Hello" />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

  test('it renders `props.text`', () => {
    const component = renderer
      .create(<Card text="World" />)
      .toJSON();

    expect(component).toMatchSnapshot();
  });

});