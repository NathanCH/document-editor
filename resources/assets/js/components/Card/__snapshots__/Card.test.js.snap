// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> it renders \`props.customSection\` 1`] = `
<div
  className="card"
>
  <span>
    Test
  </span>
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    />
    <p
      className="card-text"
    />
  </div>
</div>
`;

exports[`<Card /> it renders \`props.text\` 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    />
    <p
      className="card-text"
    >
      World
    </p>
  </div>
</div>
`;

exports[`<Card /> it renders \`props.title\` 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    >
      Hello
    </h5>
    <p
      className="card-text"
    />
  </div>
</div>
`;

exports[`<Card /> it renders an empty card 1`] = `
<div
  className="card"
>
  <div
    className="card-body"
  >
    <h5
      className="card-title"
    />
    <p
      className="card-text"
    />
  </div>
</div>
`;
